src/devices/src/virtio/block/request.rs

```diff
@@ -224,3 +224,202 @@ impl Request {
         Ok(0)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::memory_model::Address;
+    use crate::virtio::queue::tests::*;
+    use memory_model::GuestAddress;
+
+    // Writes at address 0x0 the request_type, reserved, sector.
+    fn write_request_header(mem: &GuestMemory, request_type: u32, sector: u64) {
+        let addr = GuestAddress(0);
+
+        mem.write_obj_at_addr::<u32>(request_type, addr).unwrap();
+        mem.write_obj_at_addr::<u64>(sector, addr.checked_add(8).unwrap())
+            .unwrap();
+    }
```

Comment:
We should use `RequestHeader` instead of offsets here.

Reply:
I removed this function as it was used only in one place and replaced it with a call to `mem.write_obj_at_addr::<RequestHeader>(...)`